OpenTrons/opentrons-api
# feat(protocol-designer, step-generation): add drop tip field to mix & transfer forms

closes RAUT-718

# Overview

This PR adds a drop tip field to Transfer and Mix forms and then wires it up in the pipette related compound commands in step-generation. It sorta wires stuff up in `dropTip.ts` but not fully since we don't yet know what a drop tip into waste chute will look like. This PR lays out the ground work for wiring up the `dropTip.ts` options though.

# Test Plan

- Create a Flex protocol with the deck configuration FF turned on. 
- Add a waste chute and a trash bin. 
- Go to the deck map and add a labware to the deck. Add a transfer step, see that there is a drop tip section with a location dropdown with 2 options to drop tip. See that the default is the waste chute. Add the step to drop tip in the waste chute.
- Now, create a mix step and see the drop tip section with the location dropdown again. This time, drop tip in the trash bin.
- Download the protocol and see in the json file that the 1st drop tip's labwareId is the waste chute and the 2nd drop tip's labwareId is the trash bin. NOTE: this is incorrect for the waste chute since it is not a labware but what I stubbed in for now. It will be fixed in a follow up when we know more about dropping tips in waste chute.
- Now, go back to the protocol you just created and delete either the waste chute or trash bin, go to the deck map view and see that there is a timeline error saying that the drop tip location is unknown.
- Finally, create a protocol for the OT-2, add a mix and transfer step and see that the drop tip location drop down only has the trash bin option and is automatically selected.

# Changelog

- add `dropTip_location` and `dropTipLocation` types to Transfer and Mix.
- create `DropTipField` and default the drop down location to always select something so the user won't have to select an option each time if there is only 1 option available.
- extend the args in `Transfer`, `Consolidate`, `Distribute`, and `Mix` compound commands
- create an error creator for if a drop tip is missing for some reason (which shouldn't happen unless you delete the trash bin)
- add test coverage
- add to the 7.1.0 migration to add the drop tip location fields
- fix up the cypress tests (I think some were previously failing but we can't tell since the checks are broken)

# Review requests

see test plan

# Risk assessment

low
